purchase_invoice_plan/models/purchase.py

```diff
@@ -313,7 +313,8 @@ def _get_plan_qty(self, order_line, percent):
         plan_qty = order_line.product_qty * (percent / 100)
         return plan_qty
 
-    def unlink(self):
+    @api.ondelete(at_uninstall=False)
+    def unlink_installments(self):
```

Comment:
Let's make it a private method and adjust the name, following suit to other cases I can see in the standard modules.

```suggestion
    def _unlink_except_no_edit(self):
```